File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/login/WebDavSignInView.java
package app.editors.manager.mvp.views.login;

import moxy.viewstate.strategy.OneExecutionStateStrategy;
import moxy.viewstate.strategy.StateStrategyType;

import app.editors.manager.mvp.views.base.BaseView;

@StateStrategyType(OneExecutionStateStrategy.class)
public interface WebDavSignInView extends BaseView {
    void onDialogWaiting(String string);
    void onDialogClose();
    void onLogin();
    void onUrlError(String string);
    void onNextCloudLogin(String url);
}

File: app_manager/appmanager/src/main/java/app/editors/manager/mvp/views/login/WebDavSignInView.kt
package app.editors.manager.mvp.views.login

import app.documents.core.model.login.OidcConfiguration
import app.editors.manager.mvp.views.base.BaseView
import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType

@StateStrategyType(OneExecutionStateStrategy::class)
interface WebDavSignInView : BaseView {
    fun onLogin()
    fun onDialogWaiting()
    fun onDialogClose()
    fun onUrlError()
    fun onNextCloudLogin(url: String)
    fun onOwnCloudLogin(url:String, config: OidcConfiguration?)
}